### DISCUSSION TITLE
Questions about the Source Extractor

### DISCUSSION THREAD

**Discussion by mapfiable** [sep-developers/sep#94] [Q&A] [answered]
I know the questions are not directly about `sep`, but since the original authors didn't respond to me, I was hoping that maybe you might know the answers:

- In [their paper](https://aas.aanda.org/articles/aas/pdf/1996/08/ds1060.pdf) they say "the local background histogram is clipped iteratively until convergence at ±3σ around its median". Does that mean untill all values are within ±3σ? [The guide](https://arxiv.org/pdf/astro-ph/0512139.pdf) says this more or less, however it's seemingly using mean and median interchageably ("the mean and the σ of the distribution of pixel values is computed", "[the] median and standard deviation σ are computed again", "This is repeated until all the remaining pixel values are within mean ±3σ").
- They go on to say "if σ is changed by less than 20% during that process, we consider that the field is not crowded and we simply take the mean of the clipped histogram as a value for the background". 
  - Does that mean this is a condition to stop the iteration "prematurely", i.e. not all the values of the clipped histogram necessarily lie within ±3σ at this point?
  - Why is a change of less than 20% in σ a sign that the field is not crowded? 
  - Does crowded mean crowded by (point) sources?
  - Why is this a reason to use the mean for the background value? 
- They further say that "Figure 1 shows that [their mode estimation] is considerably less affected by crowding than a simple clipped mean ... but is 30% noisier." How does the figure show that? I Can see that the clipped mean values are spread over a much lager value range than the clipped mode values, but how does that relate to the crowdedness of the sample fields? How do I know that the cropped mean values further away from the true background come from more crowded fields? How do I see that the mode estimation is noisier than the clipped mean? How is noise defined in this regard?
![Unbenannt](https://user-images.githubusercontent.com/48512334/106462035-d74d3c80-6495-11eb-8e63-cca0e7bae1a4.PNG)

**Comment by kbarbary** [comment, marked as answer]
You might have to read the code to get the answer to some of these questions. Here is where the iterative clipping is done, for example: https://github.com/kbarbary/sep/blob/master/src/background.c#L456-L491

> Why is a change of less than 20% in σ a sign that the field is not crowded?

Sounds like a pretty arbitrary threshold that the authors found "works well in practice". That would be my guess anyway. Source Extractor in general is filled with heuristics that work "pretty well" in practice, but aren't perfect.

> Does crowded mean crowded by (point) sources?

Or galaxies, yes.

> Why is this a reason to use the mean for the background value?

if you think the region of sky is empty, it's better to use mean than median or mode because it is a less noisy estimator. You can take "noisy" here to mean the expected difference between your estimate and the true value.

> They further say that "Figure 1 shows that [their mode estimation] is considerably less affected by crowding than a simple clipped mean ... but is 30% noisier." How does the figure show that?

perhaps in the region of the plot around [-2, 2], where the points are probably regions of blank sky, the clipped mode is spread over a wider range of values than the clipped median: the slope of the relation in this region is probably greater than 1.

**Reply by mapfiable** [reply]
Hey, sorry, I completely forgot to reply. Thanks a lot for contemplating those questions! Your answers definitely helped me to better understand the approach.
